### PR TITLE
fix the find_python. By default we find the `python` executable.

### PR DESCRIPTION
**Pull request by MaximilienNaveau**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

Still going back to finding python in a suitable way.

- The bug we had with @jviereck  before is that the FindPython priviledged python2
- The bug we have with @vincentberenz is that the default version of python is used but the highest version available. Hence the virtual environment is ignored.

The compilation of pybind11 still needs the `-DPYTHON_EXECUTABLE=$(which python)`

For this to work we need to find mpi_cmake_modules **before** pybind11 in our CMakeLists.txt

## How I Tested

[//]: # "Explain how you tested your changes"

I compiled everything on my laptop and checked the python bindings installations.

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [ ] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [ ] All new functions/classes are documented and existing documentation is updated according to changes.
- [ ] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

**Comment by MaximilienNaveau** [comment]
@jviereck can you try on your MacOs? So that we ensure that there are no compatibility issues with finding python for you.

**Comment by luator** [comment]
I tried building the ROBOT_FINGERS project (using an Ubuntu 20.04 container) and get the following build error:
```
CMake Error at cmake/get_python_install_dir.cmake:47 (message):
  execute_process(PYTHON_EXECUTABLE-NOTFOUND -c 'from distutils.sysconfig
  import get_python_lib;import os;install_path =
  '/home/felixwidmaier/ws/tmpws/workspace/install/mpi_cmake_modules';python_lib
  = get_python_lib(prefix=install_path);rel_path =
  os.path.relpath(python_lib,
  start=install_path);print(rel_path.replace(os.sep, '/'))') returned error
  code No such file or directory
Call Stack (most recent call first):
  CMakeLists.txt:52 (get_python_install_dir)
```

**Comment by jviereck** [comment]
I wait till the problem from Felix is figured out and then will test it on my Mac.

**Comment by MaximilienNaveau** [comment]
@luator can you try again?

**Comment by luator** [comment]
I still get the error.  Line numbers changed but I think apart from this it's the same:
```
CMake Error at cmake/get_python_install_dir.cmake:35 (message):
  execute_process(PYTHON_EXECUTABLE-NOTFOUND -c 'from distutils.sysconfig
  import get_python_lib;import os;install_path =
  '/home/felixwidmaier/ws/tmpws/workspace/install/mpi_cmake_modules';python_lib
  = get_python_lib(prefix=install_path);rel_path =
  os.path.relpath(python_lib,
  start=install_path);print(rel_path.replace(os.sep, '/'))') returned error
  code No such file or directory
Call Stack (most recent call first):
  CMakeLists.txt:52 (get_python_install_dir)
```

**Comment by MaximilienNaveau** [comment]
@luator did you clean your build?

File: cmake/find_modules/python/FindPython.cmake [hunk 1]
```diff
@@ -1,181 +1,55 @@
...
+    find_package(PythonInterp ${PYTHON_VERSION} QUIET)
   endif()
 endif()
 
-if (COMMAND __Python_add_library)
-  macro (Python_add_library)
-    __Python_add_library (Python ${ARGV})
-  endmacro()
+#
+# If the libraries and include dirs have already been found we do not for them.
```
Comment (jviereck) [inline cmake/find_modules/python/FindPython.cmake:32]: `found we do not [look] for them` (missing look?)

File: cmake/find_modules/python/FindPython.cmake.backup [hunk 2]
```diff
@@ -1,2 +1,181 @@
-find_package(PythonInterp)
-find_package(PythonLibs)
\ No newline at end of file
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
```
Comment (jviereck) [inline cmake/find_modules/python/FindPython.cmake.backup:1]: Why keeping the files as a backup? They are stored in Git anyways.

**Review by jviereck — APPROVED** [review]

**Comment by MaximilienNaveau** [comment]
The checks have been done for:
- Ubuntu 18.04
- Ubuntu 20.04
- MacOs
Let us hope that this is the last time we ever have to touch this!

**Pull request merged**